crates/bevy_pbr/src/wireframe.rs

```diff
@@ -49,31 +67,171 @@ fn extract_wireframe_config(mut commands: Commands, wireframe_config: Res<Wirefr
     }
 }
 
-fn extract_wireframes(mut commands: Commands, query: Query<Entity, With<Wireframe>>) {
-    for entity in query.iter() {
-        commands.get_or_spawn(entity).insert(Wireframe);
+fn extract_wireframes(mut commands: Commands, query: Query<(Entity, Read<Wireframe>)>) {
+    for (entity, wireframe) in query.iter() {
+        commands.get_or_spawn(entity).insert(wireframe.clone());
+    }
+}
+
+fn prepare_wireframes(
+    mut commands: Commands,
+    config: Res<WireframeConfig>,
+    render_device: Res<RenderDevice>,
+    render_queue: Res<RenderQueue>,
+    mut wireframe_meta: ResMut<GlobalWireframeMeta>,
+    mut query: QuerySet<(
+        QueryState<(Entity, Option<&Wireframe>), (With<Handle<Mesh>>, With<MeshUniform>)>,
+        QueryState<
+            (Entity, Option<&Wireframe>),
+            (With<Handle<Mesh>>, With<MeshUniform>, With<Wireframe>),
+        >,
+    )>,
```

Comment:
Why use a QuerySet her? Both Querys do not access any componenent mutably.

Reply:
I'm not too sure if it's optimal - I was mimicking `queue_wireframes`. Should it be a single query?

Reply:
These should just be two seperate queries.